sys/net/gnrc/conn/gnrc_conn.c

```diff
@@ -57,7 +73,15 @@ int gnrc_conn_recvfrom(conn_t *conn, void *data, size_t max_len, void *addr, siz
     while ((timeout--) > 0) {
         gnrc_pktsnip_t *pkt, *l3hdr;
         size_t size = 0;
-        msg_receive(&msg);
+        int idx = -1;
+        /* first check if there's packet waiting in the queue */
+        if ((idx = cib_get(&_conn_buf_cib)) != -1) {
```

Comment:
Wouldn't this also dequeue packets destined for other ports since you use only one universal cib?